python/lsst/ctrl/oods/fileAttendant.py

```diff
@@ -0,0 +1,141 @@
+# This file is part of ctrl_oods
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import shutil
+
+from lsst.ctrl.oods.butlerAttendant import ButlerAttendant
+from lsst.ctrl.oods.imageData import ImageData
+
+LOGGER = logging.getLogger(__name__)
+
+
+class FileAttendant(ButlerAttendant):
+    """Processes files on behalf of a Gen3 Butler.
+
+    Parameters
+    ----------
+    config: `dict`
+        configuration of this butler ingester
+    csc: `OodsCSC`
+        Observatory Operations Data Service Commandable SAL component
+    """
+
+    def __init__(self, config, csc=None):
+        super().__init__(config=config, csc=csc)
+
+        self.staging_dir = self.config["stagingDirectory"]
```

Comment:
It's a bit confusing to have a `stagingDirectory` and an `imageStagingDirectory`.  I think you're moving from the latter to the former, which is where things actually get ingested from (using `transfer=direct`) and is therefore a permanent location, not a staging area.

Reply:
Yes, you're correct.   It would be better to name `stagingDirectory` something else, but I'm going to leave it for now because we already have deployed systems using this, and this will soon be obsolete when we move to S3.